packages/pyright-scip/src/indexer.ts

```
@@ -123,9 +125,9 @@ export class Indexer {
         this.importResolver = new ImportResolver(fs, this.pyrightConfig, host);
 
         this.program = new Program(this.importResolver, this.pyrightConfig);
-        // Normalize paths to ensure consistency with other code paths.
-        const normalizedProjectFiles = [...this.projectFiles].map((path: string) => normalizePathCase(fs, path));
-        this.program.setTrackedFiles(normalizedProjectFiles);
+        // setTrackedFiles internally handles path normalization, so we don't normalize
+        // paths here.
+        this.program.setTrackedFiles([...this.projectFiles]);
```

This change was causing later filepaths to be normalized, so the "is path inside project root" check was failing silently on macOS. However, in that case, the test was silently passing due to incorrect testing logic.